packages/zoe/src/contractFacet/exit.js

```diff
@@ -2,65 +2,116 @@
 
 import { assert, details as X, q } from '@agoric/assert';
 import { E } from '@endo/eventual-send';
-import { Far } from '@endo/marshal';
+import { vivifyFarClass, provideDurableSetStore } from '@agoric/vat-data';
+import { M, initEmpty } from '@agoric/store';
 
 import {
   isOnDemandExitRule,
   isAfterDeadlineExitRule,
   isWaivedExitRule,
 } from '../typeGuards.js';
 
-/**
- * Makes the appropriate exitObj, which runs in ZCF and allows the seat's owner
- * to request the position be exited.
- */
+const ExitObjectGuard = M.interface('ExitObject', { exit: M.call().returns() });
+const WakerInterfaceGuard = M.interface('Waker', {
```

Comment:
Would this be better defined elsewhere, e.g. in packages/SwingSet/src/devices/timer/?

Reply:
yeah, but I doubt that the kernel will have thorough support for Patterns and Guards anytime soon, so following that path would mean either adding it all myself preemptively, or frequent, tiny changes to SwingSet in other PRs.

Reply:
```suggestion
const ExitObjectI = M.interface('ExitObject', { exit: M.call().returns() });
const WakerI = M.interface('Waker', {
```

I don't think there would be any problem using patterns and guards from the kernel, as the kernel already depends on things which depends on these. So you might want to give it a shot and proceed if you don't run into any surprises.

Reply:
Looks done. Resolving.